### ISSUE TITLE
shim 15+1533136590.3beb971-0ubuntu2 for Ubuntu

### ISSUE THREAD

**Issue by dannf** [rhboot/shim-review#71]
Make sure you have provided the following information:

 - [x] link to your code branch cloned from rhboot/shim-review in the form user/repo@tag
https://github.com/CanonicalLtd/shim-review/tree/ubuntu-shim-amd64+arm64-20190605
 - [x] completed README.md file with the necessary information
 - [x] shim.efi to be signed
 - [x] public portion of your certificate embedded in shim (the file passed to VENDOR_CERT_FILE)
 - [x] any extra patches to shim via your own git tree or as files
 - [x] any extra patches to grub via your own git tree or as files
 - [x] build logs

###### What organization or people are asking to have this signed:
Ubuntu

###### Version of shim:
shim 15-based snapshot, up to commit 3beb971

###### What product or service is this for:
Ubuntu

###### What's the justification that this really does need to be signed for the whole world to be able to boot it:
This is a well-known Linux distro.

**Comment by arrfab** [comment]
Don't know if your request is still "current" but just gave it a quick try to just build it with your supplied Dockerfile and it seems it doesn't build ? 

```
MokManager.c: In function ‘delete_keys’:
MokManager.c:1358:19: error: taking address of packed member of ‘struct <anonymous>’ may result in an unaligned pointer value [-Werror=address-of-packed-member]
 1358 |   if (CompareGuid(&(del_key[i].Type), &X509_GUID) == 0) {
      |                   ^~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [<builtin>: MokManager.o] Error 1
make: Leaving directory '/shim'
dh_auto_build: make -j4 RELEASE=15 COMMIT_ID=3beb971b10659cf78144ddc5eeea83501384440c MAKELEVEL=0 EFI_PATH=/usr/lib ENABLE_HTTPBOOT=true ENABLE_SHIM_CERT=1 ENABLE_SBSIGN=1 VENDOR_CERT_FILE=debian/canonical-uefi-ca.der EFIDIR=ubuntu returned exit code 2
make[1]: *** [debian/rules:41: override_dh_auto_build] Error 255
make[1]: Leaving directory '/shim'
make: *** [debian/rules:35: build] Error 2
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
debuild: fatal error at line 1182:
dpkg-buildpackage -us -uc -ui -i -I failed
gbp:error: 'debuild -i -I -us -uc' failed: it exited with 29
The command '/bin/sh -c gbp buildpackage -us -uc' returned a non-zero code: 1

```

**Comment by dannf** [comment]
On Fri, Sep 6, 2019 at 12:26 AM Fabian Arrotin <notifications@github.com>
wrote:

> Don't know if your request is still "current" but just gave it a quick try
> to just build it with your supplied Dockerfile and it seems it doesn't
> build ?
>

Fabian,
  Thanks for reviewing.  Yes, I can reproduce. Since the time I submitted
this review, the eoan release - which is still in devel - switched from GCC
8 to GCC 9, and this is a new check in GCC 9.

It also impacts our current signed version, so I've reported a bug here:
  https://bugs.launchpad.net/ubuntu/+source/shim/+bug/1843105

We'll need to address that, so I'll close this issue.

  -dann
